workitem/workitem_repository.go

```diff
@@ -193,12 +192,12 @@ func (r *GormWorkItemRepository) FindSecondItem(order *float64, secondItemDirect
 	Item := WorkItem{}
 	var tx *gorm.DB
 	switch secondItemDirection {
-	case DirectionBelow:
-		// Finds the item below which reorder item has to be placed
-		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
-
 	case DirectionAbove:
 		// Finds the item above which reorder item has to be placed
+		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
```

Comment:
@kwk i had tried doing that before as you had suggested. 

So, this is the query:

When direction is "**below**" in request payload
 _curl -i -X PATCH localhost:8080/api/workitems/reorder -H "Authorization: Bearer ..." -d '{"data": [{"type": "workitems", "attributes":{"version":0}, "id": "5"}], "position": {"id":"2", "direction":"below"}}'_->

then this SQL query to find second item:
**tx = r.db.Where(fmt.Sprintf("id = (SELECT max(execution_order) FROM %s WHERE (execution_order < ?))", WorkItem{}.TableName()), order).First(&Item)**

Similarly, When direction is **above**:
**tx = r.db.Where(fmt.Sprintf("id = (SELECT min(execution_order) FROM %s WHERE (execution_order > ?))", WorkItem{}.TableName()), order).First(&Item)**

When i run the same query in psql, it runs perfectly. But when i try running it through the application, it sometimes returns "0" even when it is not an edge case. M stuck here and trying to find out why this is happening.